js/apps/system/_admin/aardvark/APP/react/src/views/visgraphs/ButtonSave.js

```diff
@@ -0,0 +1,60 @@
+/* global arangoHelper, $ */
+
+import React, { useContext, useState } from "react";
+import { UrlParametersContext } from "./url-parameters-context";
+import { IconButton } from "../../components/arango/buttons";
+
+const ButtonSave = ({ graphName, onGraphDataLoaded, onIsLoadingData }) => {
+  const urlParameters = useContext(UrlParametersContext);
+  const [isLoadingData, setIsLoadingData] = useState(false)
+  let responseTimesObject = {
+    fetchStarted: null,
+    fetchFinished: null,
+    fetchDuration: null
+  }
+  const [responseTimes, setResponseTimes] = useState(responseTimesObject);
+
+  const callApi = () => {
+    localStorage.setItem(`${graphName}-gv-urlparameters`, JSON.stringify(urlParameters[0]));
+    setIsLoadingData(true);
+    onIsLoadingData(true);
+    responseTimesObject.fetchStarted = new Date();
+
+    $.ajax({
+      type: 'GET',
+      url: arangoHelper.databaseUrl(`/_admin/aardvark/visgraph/${graphName}`),
+      contentType: 'application/json',
+      data: urlParameters[0],
+      success: function (data) {
+        responseTimesObject.fetchFinished = new Date();
```

Comment:
responseTimesObject is the initalState, which is being *mutated*. Avoid mutation, move these modifications to a functions and do a shallow copy. This is true for all places. Avoid mutation as a general rule to avoid bugs